File: implementations/springboot/examples/src/main/java/com/example/orderservice/order/infrastructure/persistence/OrderJpaRepository.java
package com.example.orderservice.order.infrastructure.persistence;

import com.example.orderservice.order.domain.Order;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OrderJpaRepository extends JpaRepository<Order, Long> {
    Optional<Order> findByOrderIdAndDeletedAtIsNull(String orderId);
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/orderservice/order/infrastructure/persistence/OrderJpaRepository.kt
package com.example.orderservice.order.infrastructure.persistence

import com.example.orderservice.order.domain.Order
import org.springframework.data.jpa.repository.JpaRepository

interface OrderJpaRepository : JpaRepository<Order, Long> {
    fun findByOrderIdAndDeletedAtIsNull(orderId: String): Order?
}
